core/jsAndWasmShared/src/TimeZone.kt

```diff
@@ -19,9 +19,9 @@ public actual open class TimeZone internal constructor(internal val zoneId: Zone
     public actual fun LocalDateTime.toInstant(): Instant = toInstant(this@TimeZone)
 
     override fun equals(other: Any?): Boolean =
-            (this === other) || (other is TimeZone && this.zoneId == other.zoneId)
+            (this === other) || (other is TimeZone && (this.zoneId === other.zoneId || this.zoneId.equals(other.zoneId)))
```

Comment:
With the approach I described w.r.t. `jsTry`, this could also be hidden in the implementation of `equals` for the WASM-specific wrapper classes without polluting the shared code.